cst/plugins/eu.esdihumboldt.cst.functions.geometric/src/eu/esdihumboldt/cst/functions/geometric/join/SpatialJoinParameterType.groovy

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2013 Data Harmonisation Panel
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     Data Harmonisation Panel <http://www.dhpanel.eu>
+ */
+
+package eu.esdihumboldt.cst.functions.geometric.join;
+
+import javax.xml.parsers.DocumentBuilderFactory
+
+import org.w3c.dom.Document
+import org.w3c.dom.Element
+
+import eu.esdihumboldt.cst.functions.geometric.join.SpatialJoinParameter.SpatialJoinCondition
+import eu.esdihumboldt.hale.common.align.helper.EntityDefinitionComparator
+import eu.esdihumboldt.hale.common.align.io.LoadAlignmentContext
+import eu.esdihumboldt.hale.common.align.io.impl.DOMEntityDefinitionHelper
+import eu.esdihumboldt.hale.common.align.model.impl.TypeEntityDefinition
+import eu.esdihumboldt.hale.common.core.io.ComplexValueType
+import eu.esdihumboldt.hale.common.schema.SchemaSpaceID
+import eu.esdihumboldt.hale.common.schema.model.TypeIndex
+import eu.esdihumboldt.util.groovy.xml.NSDOMBuilder
+import eu.esdihumboldt.util.xml.XmlUtil
+import groovy.xml.dom.DOMCategory
+
+/**
+ * Descriptor for conversion between DOM and SpatialJoinParameter.
+ * 
+ * @author Kai Schwierczek
+ */
+public class SpatialJoinParameterType implements ComplexValueType<SpatialJoinParameter, LoadAlignmentContext> {
```

Comment:
As there is some dynamic code in here it would be good to have a unit test that tests the serialization and deserialization.